src/printer.rs

```diff
@@ -0,0 +1,145 @@
+use ansi_term::Style;
+use console::Term;
+use errors::*;
+use std::io::Write;
+use syntect::highlighting;
+use terminal::as_terminal_escaped;
+use {Colors, LineChange, LineChanges, Options, OptionsStyle};
+
+const PANEL_WIDTH: usize = 7;
+
+pub struct Printer<'a> {
+    handle: &'a mut Write,
+    colors: Colors,
+    term_width: usize,
+    options: &'a Options<'a>,
```

Comment:
If some of the options are only needed in the printer, we might also think about inlining them here (instead of passing the whole `Options` struct)? Not sure if that's necessarily better.

Reply:
I thought about that, but we already need 3 of the options in there, and it's likely to grow as well. So, this seemed easier